system_tests/bigtable_happybase.py

```diff
@@ -618,6 +618,197 @@ def test_scan_timestamp(self):
+        self.assertEqual(row1_after, {})
+
+    def test_delete_with_columns(self):
+        table = Config.TABLE
+        value1 = 'value1'
+        value2 = 'value2'
+        row1_data = {COL1: value1, COL2: value2}
+
+        # Need to clean-up row1 after.
+        self.rows_to_delete.append(ROW_KEY1)
+        table.put(ROW_KEY1, row1_data)
+
+        row1 = table.row(ROW_KEY1)
+        self.assertEqual(row1, row1_data)
+
+        table.delete(ROW_KEY1, columns=[COL1])
+        row1_after = table.row(ROW_KEY1)
+        self.assertEqual(row1_after, {COL2: value2})
+
+    def test_delete_with_column_family(self):
+        table = Config.TABLE
+        value1 = 'value1'
+        value2 = 'value2'
+        value3 = 'value3'
+        row1_data = {COL1: value1, COL2: value2, COL4: value3}
+
+        # Need to clean-up row1 after.
+        self.rows_to_delete.append(ROW_KEY1)
+        table.put(ROW_KEY1, row1_data)
+
+        row1 = table.row(ROW_KEY1)
+        self.assertEqual(row1, row1_data)
+
+        table.delete(ROW_KEY1, columns=[COL_FAM1])
+        row1_after = table.row(ROW_KEY1)
+        self.assertEqual(row1_after, {COL4: value3})
+
+    def test_delete_with_columns_family_overlap(self):
+        table = Config.TABLE
+        value1 = 'value1'
+        value2 = 'value2'
+        row1_data = {COL1: value1, COL2: value2}
+
+        # Need to clean-up row1 after.
+        self.rows_to_delete.append(ROW_KEY1)
+
+        # First go-around, use [COL_FAM1, COL1]
+        table.put(ROW_KEY1, row1_data)
+        row1 = table.row(ROW_KEY1)
+        self.assertEqual(row1, row1_data)
+
+        table.delete(ROW_KEY1, columns=[COL_FAM1, COL1])
+        row1_after = table.row(ROW_KEY1)
+        self.assertEqual(row1_after, {})
+
+        # Second go-around, use [COL1, COL_FAM1]
+        table.put(ROW_KEY1, row1_data)
+        row1 = table.row(ROW_KEY1)
+        self.assertEqual(row1, row1_data)
+
+        table.delete(ROW_KEY1, columns=[COL1, COL_FAM1])
+        row1_after = table.row(ROW_KEY1)
+        self.assertEqual(row1_after, {})
+
+    def test_delete_with_timestamp(self):
+        table = Config.TABLE
+        value1 = 'value1'
+        value2 = 'value2'
+
+        # Need to clean-up row1 after.
+        self.rows_to_delete.append(ROW_KEY1)
+        table.put(ROW_KEY1, {COL1: value1})
+        table.put(ROW_KEY1, {COL2: value2})
+
+        row1 = table.row(ROW_KEY1, include_timestamp=True)
+        ts1 = row1[COL1][1]
+        ts2 = row1[COL2][1]
+
+        self.assertTrue(ts1 < ts2)
+
+        # NOTE: The Cloud Bigtable "Mutation.DeleteFromRow" mutation does
+        #       not support timestamps. Even attempting to send one
+        #       conditionally(via CheckAndMutateRowRequest) deletes the
+        #       entire row.
+        # NOTE: Cloud Bigtable deletes **ALSO** use an inclusive timestamp
+        #       at the endpoint, but only because we fake this when
+        #       creating Batch._delete_range.
```

Comment:
Sorry, they are a bit out of context.

In the tests [in `gcloud-python-bigtable`][1], there is a switch to run these against the Cloud Bigtable backend or the HBase/Thrift backend. This comment was to describe why they are a bit different.

I left them in because

1. I plan on sending a follow-up to add in the HBase/Thrift switch (still may be worthwhile)
1. It's worth noting points where HBase/Thrift and Cloud Bigtable differ in behavior

[1]: https://github.com/dhermes/gcloud-python-bigtable/blob/d3789abbb95a97f3debc09955aa53c06d1871fcf/system_tests/run_happybase.py#L879